docs/src/index.md

```diff
@@ -2,10 +2,8 @@
 
-The package provides the `CategoricalArray` type designed to hold categorical data (either unordered/nominal or ordered/ordinal) efficiently and conveniently. `CategoricalArray{T}` holds values of type `T`. `CategoricalArray{Union{Null, T}}`, either holds values of type `T`, or missing values (represented as `null`, of the [`Null`](https://github.com/JuliaData/Nulls.jl) type). When indexed, `CategoricalArray{T}` returns values of type `CategoricalValue{T}` rather than of type `T` (or `null` for missing values).
+The package provides the `CategoricalArray` type designed to hold categorical data (either unordered/nominal or ordered/ordinal) efficiently and conveniently. `CategoricalArray{T}` holds values of type `T`. The `CategoricalArray{Union{Null, T}}` variant can also contain missing values (represented as `null`, of the [`Null`](https://github.com/JuliaData/Nulls.jl) type). When indexed, `CategoricalArray{T}` returns special "categorical value" objects (`CategoricalString` for `T = String` or `CategoricalValue{T}` for any other `T`) rather than the original values of type `T`. `CategoricalString` and `CategoricalValue` are simple wrappers around the categorical levels; these types allow very efficient retrieval and comparison of actual values.
 
-`CategoricalValue` objects are simple wrappers around the actual categorical levels which allow for very efficient extraction and equality tests. Indeed, the main feature of categorical arrays types is that they store a pool of the levels which can appear in the variable. These levels are stored in a specific order: for unordered arrays, this order is only used for pretty printing (e.g. in cross tables or plots); for ordered arrays, it also allows comparing values using the `<` and `>` operators: the comparison is then based on the ordering of levels stored in the array. Whether an array is ordered can be defined either on construction via the `ordered` argument, or at any time via the `ordered!` function.
+Indeed, the main feature of `CategoricalArray` is that it maintains a pool of the levels which can appear in the data. These levels are stored in a specific order: for unordered arrays, this order is only used for pretty printing (e.g. in cross tables or plots); for ordered arrays, it also allows comparing values using the `<` and `>` operators: the comparison is then based on the ordering of levels stored in the array. Whether an array is ordered can be defined either on construction via the `ordered` argument, or at any time via the `ordered!` function. The `levels` function returns all the levels of `CategoricalArray`, and `levels!` function can be used to set the levels and their order. Levels are also automatically created when setting an array element to a level not encountered before. But they are never removed without manual intervention: use the `droplevels!` function for this.
```

Comment:
My comment still holds here: "the `levels!` function"; "Levels are also automatically expanded".

Reply:
I'm torn between "expanded" and "extended". [These tips](https://www.dailywritingtips.com/expanded-and-extended/) do not help the decision much. :)
Will use "expanded".

Reply:
Hmm, "extended" sounds more correct. Indeed we change `length(levels(x))`, not `area(levels(x))`. :-)